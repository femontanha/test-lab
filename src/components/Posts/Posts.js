import React, { PureComponent } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import styles from './posts.css';

const propTypes = {
  posts: PropTypes.object.isRequired,
};

class Posts extends PureComponent {
  renderPosts() {
    const { posts } = this.props;

    return Object.keys(posts).map((item) => (
      <article className={ styles.article } key={ posts[item].id }>
        <h1 className={ styles.title }>{ posts[item].title }</h1>
        <p className={ styles.text }>{ posts[item].body }</p>
      </article>
    ));
  }

  render() {
    return (
      <div className={ styles.posts }>
        <Link className="app-button" to="users">View Users</Link>
        { this.renderPosts() }
      </div>
    );
  }
}

Posts.propTypes = propTypes;

export default Posts;
