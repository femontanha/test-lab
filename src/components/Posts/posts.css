.posts {
  margin: 0 auto;
  max-width: 900px;
  text-align: center;
  padding: 30px;
}

.article {
  margin-top: 30px;
}

.title {
  color: var(--post-title-color);
  font-size: 22px;
  margin: 0 0 10px;
}

.text {
  margin: 0;
}
