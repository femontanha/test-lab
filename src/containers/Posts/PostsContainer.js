import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { fetchPosts } from '../../actions/posts';

class PostsContainer extends PureComponent {
  componentDidMount() {
    this.props.fetchPosts();
  }

  render() {
    console.log('this.props', this.props.posts)
    return (
      <div>Teste</div>
    );
  }
}

const mapStateToProps = state => {
  return {
    posts: state.posts,
  }
};

const mapDispatchToProps = {
  fetchPosts,
}

export default connect(mapStateToProps, mapDispatchToProps)(PostsContainer);
