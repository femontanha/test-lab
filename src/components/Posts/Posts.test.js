import React from 'react';
import Posts from './Posts';
import renderer from 'react-test-renderer';

test('render posts', () => {
  const posts = {
    '0': {
      body: 'Body',
      id: 1,
      title: 'Title',
      userId: 1,
    }
  }

  const component = renderer.create(
    <Posts posts={ posts } />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
