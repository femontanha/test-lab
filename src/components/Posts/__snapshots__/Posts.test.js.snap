// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render posts 1`] = `
<div
  className={undefined}
>
  <article
    className={undefined}
  >
    <h1
      className={undefined}
    >
      Title
    </h1>
    <p
      className={undefined}
    >
      Body
    </p>
  </article>
</div>
`;
